lib/chusaku.rb

```diff
@@ -19,14 +21,17 @@ def call(flags = {})
       @flags = flags
       @routes = Chusaku::Routes.call
       @changed_files = []
-      controllers_pattern = @flags[:controllers_pattern] || "app/controllers/**/*_controller.rb"
+      controllers_pattern = @flags[:controllers_pattern] || DEFAULT_CONTROLLERS_PATTERN
+      controllers_paths = Dir.glob(Rails.root.join(controllers_pattern))
+
+      @routes.each do |controller, actions|
+        next unless controller
 
-      Dir.glob(Rails.root.join(controllers_pattern)).each do |path|
-        controller = %r{controllers/(.*)_controller\.rb}.match(path)[1]
-        actions = @routes[controller]
-        next if actions.nil?
+        controller_class = "#{controller.underscore.camelize}Controller".constantize
+        source_path = controller_class.instance_method(actions.keys.first.to_sym).source_location[0]
+        next unless controllers_paths.include?(source_path)
```

Comment:
Approach-wise, this is great. I like this method of asking the controller class directly where the source lives. As you mentioned, it definitely gets around the current issue of being unable to find non-conventional controller paths.

I'm now realizing it's potentially possible for `source_path` to _not_ show up in `controllers_paths` even if it defaults to `Chusaku::DEFAULT_CONTROLLERS_PATTERN`, which feels problematic. I'm wondering if the guard should be a little looser here. Perhaps just checking that `source_path` exists at all is a better conditional than what I originally put here. What do you think?

Reply:
You're right, it is possible for `source_path` to be outside of `controller_paths`, especially if you use a gem which is also an engine. But I think that's why we need some restriction. Otherwise, this would modify the source of gems, since they are available on the filesystem.

We could loosen it up a bit, by checking that the path is within `Rails.root`. Something like `DEFAULT_CONTROLLERS_PATTERN = "**/*_controller.rb"`. That would mean most apps (including my case of embedded engines) don't need to customize anything.

What do you think?

Reply:
Ah, that's a good point! I didn't consider local gems getting modified with my suggested approach. I like defaulting `DEFAULT_CONTROLLERS_PATTERN` to what you suggested here—that feels the most right so far.